docs/examples/algo/complete-example.py

```diff
@@ -1,87 +1,106 @@
 from nupic.research.spatial_pooler import SpatialPooler
 from nupic.research.temporal_memory import TemporalMemory
 from nupic.algorithms.sdr_classifier_factory import SDRClassifierFactory
 
-_INPUT_FILE_PATH = "../data/gymdata.csv"
+_NUM_RECORDS = 3000
+_EXAMPLE_DIR = os.path.dirname(os.path.abspath(__file__))
+_INPUT_FILE_PATH = os.path.join(_EXAMPLE_DIR, os.pardir, "data", "gymdata.csv")
+_PARAMS_PATH = os.path.join(_EXAMPLE_DIR, os.pardir, "params", "model.yaml")
 
-def runHotgym():
 
-  timeOfDayEncoder = DateEncoder(timeOfDay=(21,1))
-  weekendEncoder = DateEncoder(weekend=21)
-  scalarEncoder = RandomDistributedScalarEncoder(0.88)
 
-  encodingWidth = timeOfDayEncoder.getWidth() \
-    + weekendEncoder.getWidth() \
-    + scalarEncoder.getWidth()
+def runHotgym():
+  with open(_PARAMS_PATH, "r") as f:
+    modelParams = yaml.safe_load(f)["modelParams"]
+    enParams = modelParams["sensorParams"]["encoders"]
+    spParams = modelParams["spParams"]
+    tmParams = modelParams["tmParams"]
+
+  timeOfDayEncoder = DateEncoder(
+    timeOfDay=enParams["timestamp_timeOfDay"]["timeOfDay"])
+  weekendEncoder = DateEncoder(
+    weekend=enParams["timestamp_weekend"]["weekend"])
+  scalarEncoder = RandomDistributedScalarEncoder(
+    enParams["consumption"]["resolution"])
+
+  encodingWidth = (timeOfDayEncoder.getWidth()
+                   + weekendEncoder.getWidth()
+                   + scalarEncoder.getWidth())
 
   sp = SpatialPooler(
     # How large the input encoding will be.
     inputDimensions=(encodingWidth),
     # How many mini-columns will be in the Spatial Pooler.
-    columnDimensions=(2048),
-    # What percent of the columns's receptive field is available for potential
+    columnDimensions=(spParams["columnCount"]),
+    # What percent of the columns"s receptive field is available for potential
     # synapses?
-    potentialPct=0.85,
+    potentialPct=spParams["potentialPct"],
     # This means that the input space has no topology.
-    globalInhibition=True,
-    localAreaDensity=-1.0,
+    globalInhibition=spParams["globalInhibition"],
+    localAreaDensity=spParams["localAreaDensity"],
     # Roughly 2%, giving that there is only one inhibition area because we have
     # turned on globalInhibition (40 / 2048 = 0.0195)
-    numActiveColumnsPerInhArea=40.0,
+    numActiveColumnsPerInhArea=spParams["numActiveColumnsPerInhArea"],
     # How quickly synapses grow and degrade.
-    synPermInactiveDec=0.005,
-    synPermActiveInc=0.04,
-    synPermConnected=0.1,
+    synPermInactiveDec=spParams["synPermInactiveDec"],
+    synPermActiveInc=spParams["synPermActiveInc"],
+    synPermConnected=spParams["synPermConnected"],
     # boostStrength controls the strength of boosting. Boosting encourages
     # efficient usage of SP columns.
-    boostStrength=3.0,
+    boostStrength=spParams["boostStrength"],
     # Random number generator seed.
-    seed=1956,
+    seed=spParams["seed"],
+    # TODO: is this useful?
```

Comment:
@marionleborgne If you choose not to make the change above, let me know so I can merge as-is.